core/src/main/java/io/github/tcdl/msb/collector/CollectorManager.java

```diff
@@ -46,32 +45,30 @@ public void handleMessage(Message message) {
 
     /**
      * @throws ConsumerSubscriptionException if another consumer already listen for messages on topic
-     * @throws DuplicateCollectorException   if another collector with the same collectorId is already registered for messages on topic
      */
-    public synchronized void registerCollector(Collector collector) {
+    public void registerCollector(Collector collector) {
         String correlationId = collector.getRequestMessage().getCorrelationId();
-        Collector newCollector = collectorsByCorrelationId.putIfAbsent(correlationId, collector);
+        collectorsByCorrelationId.putIfAbsent(correlationId, collector);
 
-        // if there was no collector with same CorrelationId
-        if (newCollector == null) {
+        synchronized (this) {
```

Comment:
this change seems to be not relevant to tags. Let's try to keep separate things in separates PRs. I wouldn't mind if you keep it for now.